backend/core.go

```diff
@@ -97,4 +97,5 @@ type CallResourceHandler interface {
 // QueryDataHandler handles data queries.
 type QueryDataHandler interface {
 	QueryData(ctx context.Context, req *QueryDataRequest) (*QueryDataResponse, error)
+	HealthCheckDatasource(ctx context.Context, req *DataSourceConfig) (*CheckHealthResult, error)
```

Comment:
Perhaps we should pass the whole PluginConfig here?
It won't be used now but maybe in the future.